pkg/apis/build/v1alpha1/buildstrategy.go

```diff
@@ -68,7 +68,7 @@ type Strategy struct {
 
 	// API version of the referent
 	// +optional
-	APIVersion string `json:"apiVersion,omitempty"`
+	APIVersion *string `json:"apiVersion,omitempty"`
```

Comment:
I think ditto from @SaschaSchwarze0's earlier comments.

I also wonder if `Kind` needs to be required so we distinguish `BuildStrategy` from `ClusterBuildStrategy`

Reply:
As of now if the user didn't provide any values for `Kind` we default to the namespaced-scope strategy.  If we make the `Kind` field required we can remove this check. See [here](https://github.com/shipwright-io/build/blob/e558e14c5d0e110290a08dce63514a48a87281f7/pkg/reconciler/buildrun/buildrun.go#L421).

We are already updating an error on invalid `Kind` value.